signac/contrib/schema.py

```diff
@@ -9,19 +11,55 @@
 
 
 class _Vividict(dict):
+    """A dict that returns an empty _Vividict for keys that are missing.
+
+    Useful for automatically nesting keys with ``vividict['a']['b']['c'] = 'd'``.
+
+    """
     def __missing__(self, key):
         value = self[key] = type(self)()
         return value
 
 
 def _collect_by_type(values):
+    """Construct a mapping of types to a set of elements drawn from the input values.
+
+    Parameters
+    ----------
+    values :
+        An iterable of values.
+
+    Returns
+    -------
+    defaultdict(set)
+        A mapping of types to a set of input values of that type.
+
+    """
     values_by_type = ddict(set)
     for v in values:
         values_by_type[type(v)].add(v)
     return values_by_type
 
 
 def _build_job_statepoint_index(jobs, exclude_const, index):
```

Comment:
The `jobs` argument isn't used here. Since this is a private method, this argument can be removed. I pushed a commit that removed it and updated the docstrings for this method.

Reply:
thanks!